src/common/src/array/list_array.rs

```diff
@@ -348,11 +348,31 @@ impl ListValue {
         datatype: &DataType,
         deserializer: &mut memcomparable::Deserializer<impl Buf>,
     ) -> memcomparable::Result<Self> {
-        let len = u32::deserialize(&mut *deserializer)?;
-        (0..len)
-            .map(|_| deserialize_datum_from(datatype, deserializer))
-            .try_collect()
-            .map(Self::new)
+        // This is a bit dirty, but idk how to correctly deserialize bytes in memcomparable
+        // format without this...
+        struct Visitor<'a>(&'a DataType);
+        impl<'a> serde::de::Visitor<'a> for Visitor<'a> {
+            type Value = Vec<u8>;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                write!(formatter, "a list of {}", self.0)
+            }
+
+            fn visit_byte_buf<E>(self, v: Vec<u8>) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(v)
+            }
+        }
+        let visitor = Visitor(datatype);
+        let bytes = deserializer.deserialize_byte_buf(visitor)?;
```

Comment:
So I found this tentative solution:
```
let bytes: &[u8] = Deserialize::deserialize(deserializer)?;
```

However it is reporting error:
```
invalid type: byte array, expected a borrowed byte array
```

Internally there is a [builtin](https://github.com/serde-rs/serde/blob/3d0251666e2af77091fff0d44f4b2eb127c9487b/serde/src/de/impls.rs#L634-L665) `BytesVisitor`, which only implements `visit_borrowed_bytes` but not `visit_bytes`.

While the [comment](https://github.com/serde-rs/serde/blob/3d0251666e2af77091fff0d44f4b2eb127c9487b/serde/src/de/mod.rs#L1549-L1550) of `visit_bytes` says:
> It is never correct to implement `visit_byte_buf` without implementing `visit_bytes`. Implement neither, both, or just `visit_bytes`.

Not sure if this is a bug in the serde library ...

Reply:
Yep, so I choose to define a Visitor myself....